### PR TITLE
Css inline changes

### PR DESCRIPTION
**Pull request by pavanmt9**
GMAIL is not rendering properly the html contents. Hence added this logic to support envInjection in jenkins to support html even in case DATA_INLINE_ATTR is not set in html reports.

### CODE REVIEW

**Comment by slide** [comment]
If rather not use a system property. A configuration option would be better.

**Comment by pavanmt9** [comment]
Sure, thanks for the suggestion. I will do the changes and submit the pull request again.

Probably I will add a checkbox with default disabled, then will validate the value and turn on the logic. Hope this implementation should be fine?

Enable CSS Inline data attribute.

**Comment by slide** [comment]
Yes, that sounds good. Thanks.

**Comment by pavanmt9** [comment]
@slide 

Please review if these changes are fine. As you had suggested, implemented enabling css inline as a configuration param.

 Thanks

File: src/main/java/hudson/plugins/emailext/ExtendedEmailPublisher.java [hunk 1]
```diff
@@ -159,14 +164,14 @@ public ExtendedEmailPublisher(String project_recipient_list, String project_cont
 
         this(project_recipient_list, project_content_type, project_default_subject, project_default_content,
                 project_attachments, project_presend_script, project_attach_buildlog, project_replyto,
-                project_save_output, project_triggers, matrixTriggerMode, false, Collections.EMPTY_LIST);
+                project_save_output, project_triggers, matrixTriggerMode, false,false, Collections.EMPTY_LIST);
```
Comment (slide) [inline src/main/java/hudson/plugins/emailext/ExtendedEmailPublisher.java]: Please use a DataBoundSetter for any new parameters.

File: src/main/java/hudson/plugins/emailext/ExtendedEmailPublisher.java [hunk 2]
```diff
@@ -159,14 +164,14 @@ public ExtendedEmailPublisher(String project_recipient_list, String project_cont
...
     }
 
     @DataBoundConstructor
     public ExtendedEmailPublisher(String project_recipient_list, String project_content_type, String project_default_subject,
             String project_default_content, String project_attachments, String project_presend_script,
             int project_attach_buildlog, String project_replyto, boolean project_save_output,
-            List<EmailTrigger> project_triggers, MatrixTriggerMode matrixTriggerMode, boolean project_disabled,
+            List<EmailTrigger> project_triggers, MatrixTriggerMode matrixTriggerMode, boolean project_disabled,boolean css_inline_enabled,
```
Comment (slide) [inline src/main/java/hudson/plugins/emailext/ExtendedEmailPublisher.java]: Sorry, this is where you should use the DataBoundSetter

File: src/main/java/hudson/plugins/emailext/ExtendedEmailPublisher.java [hunk 3]
```diff
@@ -735,7 +745,7 @@ private Multipart addContent(ExtendedEmailPublisherContext context, String chars
         messageContentType += "; charset=" + charset;
 
         try {
-            if (saveOutput) {
+			if (saveOutput) {
```
Comment (slide) [inline src/main/java/hudson/plugins/emailext/ExtendedEmailPublisher.java]: Whitespace change?

File: src/main/java/hudson/plugins/emailext/ExtendedEmailPublisher.java [hunk 4]
```diff
@@ -763,8 +773,27 @@ private Multipart addContent(ExtendedEmailPublisherContext context, String chars
...
+            /**Adding the code to support gmail integration without relying on DATA_INLINE_ATTR check. 
+            *  Most of the reports doesnot have this attribute. 
+            * 
+            */
+            
+           if(cssInlineEnable){
+            	
+            	inlinedCssHtml=inliner.processCssInline(text);
+            	
+            }else{
```
Comment (slide) [inline src/main/java/hudson/plugins/emailext/ExtendedEmailPublisher.java]: Whitespace issues around this area

**Comment by pavanmt9** [comment]
@slide 

Thanks a lot. I have made the changes as you had suggested. Please review.

**Comment by pavanmt9** [comment]
@slide 

Can you please review the changes.

**Comment by pavanmt9** [comment]
@slide

Can you please review the changes.

File: src/main/java/hudson/plugins/emailext/plugins/CssInliner.java [hunk 5]
```diff
@@ -96,6 +96,44 @@ public String process(String input) {
...
+        extractStyles(doc);
+        applyStyles(doc);
+        inlineImages(doc);
+
+        doc.outputSettings(doc.outputSettings().prettyPrint(false).escapeMode(Entities.EscapeMode.xhtml));
+        return StringEscapeUtils.unescapeHtml(doc.outerHtml());
+    }
+    
+    
+    public String process(String input,boolean dataInlinerCheck) {
```
Comment (slide) [inline src/main/java/hudson/plugins/emailext/plugins/CssInliner.java]: You never use the dataInlinerCheck parameter

File: src/main/java/hudson/plugins/emailext/plugins/CssInliner.java [hunk 6]
```diff
@@ -96,6 +96,44 @@ public String process(String input) {
         doc.outputSettings(doc.outputSettings().prettyPrint(false).escapeMode(Entities.EscapeMode.xhtml));
         return StringEscapeUtils.unescapeHtml(doc.outerHtml());
     }
+    
+    public String processCssInline(String input) {
+        Document doc = Jsoup.parse(input);
+
+        // check if the user wants to inline the data
+       /**
```
Comment (slide) [inline src/main/java/hudson/plugins/emailext/plugins/CssInliner.java]: Don't leave commented code

File: src/main/resources/hudson/plugins/emailext/ExtendedEmailPublisher/config.groovy [hunk 7]
```diff
@@ -14,6 +14,10 @@ f.entry(title: _("Disable Extended Email Publisher"), help: "/plugin/email-ext/h
     f.checkbox(name: "project_disabled", checked: instance?.disabled)
 }
 
+f.entry(title: _("Enable CSS Inline Attribute"), help: "/plugin/email-ext/help/projectConfig/cssInlineEnable.html", description: _("Allows html css inline rendering on gmail webclients")) {
```
Comment (slide) [inline src/main/resources/hudson/plugins/emailext/ExtendedEmailPublisher/config.groovy]: Where is this used?

**Comment by slide** [comment]
You did something that made too many changes in one file, ExtendedEmailPublisher.java. 808 additions, 781 removals...I can't merge something like that.

**Pull request closed without merging**